backend-tests/tests/test_create_artifact.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import requests
+import time
+
+from json import dumps
+
+from testutils.api.client import ApiClient
+from testutils.api import deployments, useradm
+from testutils.common import mongo, clean_mongo, create_org
+
+
+def test_create_artifact(mongo, clean_mongo):
+    tenant, username, password = "test.mender.io", "user@test.mender.io", "secret"
+    tenant = create_org(tenant, username, password)
+    r = ApiClient(useradm.URL_MGMT).call(
+        "POST", useradm.URL_LOGIN, auth=(username, password)
+    )
+    assert r.status_code == 200
+    assert r.text is not None
+    assert r.text != ""
+
+    filename = "/tests/run.sh"
+
+    auth_token = r.text
+    api_client = ApiClient(deployments.URL_MGMT)
+    r = requests.post(
+        deployments.URL_MGMT + deployments.URL_DEPLOYMENTS_ARTIFACTS_GENERATE,
+        headers={"Authorization": "Bearer " + auth_token},
+        files={
+            "name": (None, "artifact"),
+            "description": (None, "description"),
+            "device_types_compatible": (None, "beaglebone"),
+            "type": (None, "single_file"),
+            "args": (None, dumps({"filename": "run.sh", "dest_dir": "/tests",})),
+            "file": (filename, open(filename, "rb"), "application/octet-stream", {}),
+        },
+        verify=False,
+    )
```

Comment:
As it is now, `api_client` is unused.

In other bakend-tests I have seen the requests expressed in the following manner:
```
api_client.with_auth(auth_token).call("POST", deployments.URL_DEPLOYMENTS_ARTIFACTS_GENERATE, files=...)
```

Reply:
Right, I'm changing the code.